Sources/Error.swift

```diff
@@ -2,22 +2,26 @@ public enum Error: ErrorType, Equatable {
     case General
     case InvalidUrl(String)
     case InvalidResponseData(NSData?)
-    case Error(NSError?)
+    case ErrorNS(NSError?)
+    case Error(ErrorType)
     case InvalidAuthentication
     case ShouldOverride
+    case CreateDataTask
```

Comment:
I had an error when the dataTask returned nil and then the app dit not work. This is why I also put the guard in. Another option is to force unwrap but this would crash the app. That is why I added it.